src/PowerSystems2PRAS.jl

```diff
@@ -537,27 +435,175 @@ function generate_pras_system(
-    #######################################################
-    @info "Processing GeneratorStorages in PSY System... "
-
-    gen_stor = []
-    for i in 1:num_regions
-        if (length(gen_stors[i]) != 0)
-            push!(gen_stor, gen_stors[i]...)
-        end
-    end
+"""
+    $(TYPEDSIGNATURES)
 
-    if (length(gen_stor) == 0)
-        gen_stor_names = String[]
+Apply HybridSystem Formulation to fill in a row of a PRAS Matrix.
+Views should be passed in for all arrays.
+"""
+function assign_to_gen_stor_matrices!(
+    formulation::HybridSystemFormulation,
+    g_s::PSY.Device,
+    s2p_meta::S2P_metadata,
+    charge_cap_array,
+    discharge_cap_array,
+    inflow_array,
+    energy_cap_array,
+    gridinj_cap_array,
+)
+    fill!(
+        charge_cap_array,
+        floor(Int, PSY.get_input_active_power_limits(PSY.get_storage(g_s)).max),
+    )
+    fill!(
+        discharge_cap_array,
+        floor(Int, PSY.get_output_active_power_limits(PSY.get_storage(g_s)).max),
+    )
+    fill!(
+        energy_cap_array,
+        floor(Int, PSY.get_state_of_charge_limits(PSY.get_storage(g_s)).max),
+    )
+    fill!(gridinj_cap_array, floor(Int, PSY.get_output_active_power_limits(g_s).max))
+
+    if (
+        PSY.has_time_series(PSY.get_renewable_unit(g_s)) && (
+            formulation.max_active_power in
+            PSY.get_name.(
+                PSY.get_time_series_multiple(
+                    PSY.get_renewable_unit(g_s),
+                    s2p_meta.filter_func,
+                )
+            )
+        )
+    )
+        inflow_array .=
+            floor.(
+                Int,
+                get_ts_values(
+                    get_first_ts(
+                        PSY.get_time_series_multiple(
+                            PSY.get_renewable_unit(g_s),
+                            s2p_meta.filter_func,
+                            name=formulation.max_active_power,
+                        ),
+                    ),
+                ) * PSY.get_max_active_power(PSY.get_renewable_unit(g_s)),
+            )
     else
-        gen_stor_names = PSY.get_name.(gen_stor)
+        fill!(
+            inflow_array,
+            floor(Int, PSY.get_max_active_power(PSY.get_renewable_unit(g_s))),
+        )
     end
+end
 
-    gen_stor_categories = string.(typeof.(gen_stor))
+"""
+    $(TYPEDSIGNATURES)
+
+Apply HydroEnergyReservoir Formulation to fill in a row of a PRAS Matrix.
+Views should be passed in for all arrays.
+"""
+function assign_to_gen_stor_matrices!(
+    formulation::HydroEnergyReservoirFormulation,
+    g_s::PSY.Device,
+    s2p_meta::S2P_metadata,
+    charge_cap_array,
+    discharge_cap_array,
+    inflow_array,
+    energy_cap_array,
+    gridinj_cap_array,
+)
+    if (PSY.has_time_series(g_s))
+        if (
+            formulation.inflow in
+            PSY.get_name.(PSY.get_time_series_multiple(g_s, s2p_meta.filter_func))
+        )
+            charge_cap_array .=
```

Comment:
These should be moved to separate functions too

Reply:
Done.